upcoming/composable-pipelines-pattern.md

```diff
@@ -0,0 +1,392 @@
++++
+Author = ["Samuel Lampa"]
+date = "2015-01-30T00:00:00-08:00"
+series = [""]
+title = "Patterns for composable concurrent pipelines in Go"
++++
+
+# Patterns for composable concurrent pipelines in Go
+
+## Generator function in Python
+
+The author of this post came into Go from the python-world, which is quite prevalent the field of bioinformatics. In python one can easily write composable lazy-evauated pipelines of string processing operations using the generator syntax built into the language, which has been shown to both use less memory and to be a bit faster than their eager counterparts.
+
+The Generator functionality in python basically means that you create a function that, rather than returning a single data structure once (say for example a list of items), rather will return a generator object which can later be iterated over by repeadetly calling its next method, or by the shorthand `for item in ...` syntax.
+
+What is special about a generator compared to other *iterables* in python such as lists (which also support the `for item in ...` syntax), is that the generator function will start evaluating itself and yielding its output objects one by one only after iteration has started. Thus, minimal intermediate data is created and stored in memory.
+
+To give an example, say that we have a file, chr_y.fa containing a little bit of the familiar A, C, G, T DNA nucleotides from the human Y chromosome, in the ubiquotous [FASTA file format](http://en.wikipedia.org/wiki/FASTA_format):
```

Comment:
Re-wrote the explanation, to be more explicit, and link to some wikipedia articles.